UnitTests/UI/Test_AdvancedCollectionView.cs

```diff
@@ -70,6 +70,117 @@ public void Test_AdvancedCollectionView_Filter()
 
         [TestCategory("AdvancedCollectionView")]
         [UITestMethod]
+        public void Test_AdvancedCollectionView_FilterOrder()
+        {
+            var l = new ObservableCollection<string>
+            {
+                "lorem",
+                "ipsum",
+                "dolor",
+                "sit",
+                "amet"
+            };
+
+            var a = new AdvancedCollectionView(l)
+            {
+                Filter = (x) => x.ToString().Length < 5
+            };
+
+            a.Insert(4, "how");
+
+            Assert.AreEqual(3, a.Count);
+            Assert.AreEqual("sit", a[0]);
+            Assert.AreEqual("how", a[1]);
+            Assert.AreEqual("amet", a[2]);
+
+        }
+
+        [TestCategory("AdvancedCollectionView")]
+        [UITestMethod]
+        public void Test_AdvancedCollectionView_FilterInsertDuplicate()
+        {
+            var l = new ObservableCollection<string>
+            {
+                "lorem",
+                "ipsum",
+                "dolor",
+                "sit",
+                "amet"
+            };
+
+            var a = new AdvancedCollectionView(l)
+            {
+                Filter = (x) => x.ToString().Length >= 5
+            };
+
+            a.Insert(3, "ipsum");
+
+            Assert.AreEqual(4, a.Count);
+            Assert.AreEqual(4, a.Count);
+            Assert.AreEqual("lorem", a[0]);
+            Assert.AreEqual("ipsum", a[1]);
+            Assert.AreEqual("dolor", a[2]);
+            Assert.AreEqual("ipsum", a[3]);
+
+        }
+
+        [TestCategory("AdvancedCollectionView")]
+        [UITestMethod]
+        public void Test_AdvancedCollectionView_RefreshFilterWithDuplicate()
+        {
+            var l = new ObservableCollection<string>
+            {
+                "lorem",
+                "ipsum",
+                "dolor",
+                "sit",
+                "ipsum",
+                "amet"
+            };
+
+            var a = new AdvancedCollectionView(l)
+            {
+                Filter = (x) => x.ToString().Length < 5
+            };
+
+            a.Filter = (x) => x.ToString().Length >= 5;
+
+            Assert.AreEqual(4, a.Count);
+            Assert.AreEqual("lorem", a[0]);
+            Assert.AreEqual("ipsum", a[1]);
+            Assert.AreEqual("dolor", a[2]);
+            Assert.AreEqual("ipsum", a[3]);
+
+        }
+
+        [TestCategory("AdvancedCollectionView")]
+        [UITestMethod]
+        public void Test_AdvancedCollectionView_FilterOrderShowBefore()
```

Comment:
Is this just a different order of insertion from the first test?  Maybe add a comment with intent?

Reply:
I went for a new test name altogether, let me know if you think it is clear enough without a comment now.